x/gamm/pool-models/balancer/amm_joinpool_test.go

```diff
@@ -786,7 +814,12 @@ func TestCalcSingleAssetJoin(t *testing.T) {
 				}
 
 				require.NoError(t, err)
-				assertExpectedSharesErrRatio(t, tc.expectShares, shares)
+				err = assertExpectedSharesErrRatio(t, tc.expectShares, shares)
+				if tc.expectToleranceError {
```

Comment:
I don't see a single test case where `expectToleranceError` is true, do we need to have this conditional?

Reply:
i have added two tests so far; 
1. https://github.com/osmosis-labs/osmosis/pull/1822/files#diff-d90d7fb125be0247c48c167de4adedf2ddbdc76b1ad06bc243d329bc8dbff76dR985
2. https://github.com/osmosis-labs/osmosis/pull/1822/files#diff-d90d7fb125be0247c48c167de4adedf2ddbdc76b1ad06bc243d329bc8dbff76dR997

Reply:
But these are defined in a different test fixture - `TestCalcJoinSingleAssetTokensIn`, are they not?

The test cases for this particular fixture (`TestCalcJoinPoolShares`) seem to be unmodified

Reply:
aaah i see! I thought i was only creating tolerance check tests for `TestCalcJoinSingleAssetTokensIn` function. I can create tests for `TestCalcJoinPoolShares` too